src/app/doctor/doctor.hpp

```diff
@@ -1,16 +1,19 @@
 #pragma once
 
 #include "../person/person.hpp"
-#include <set>
+#include <vector>
 
 class Visit;
 
-class Doctor : public Person, public std::enable_shared_from_this<Doctor>
+class Doctor : public Person
 {
-    std::set<std::shared_ptr<Visit>> visit_associations_;
+    std::vector<std::shared_ptr<Visit>> visits_;
```

Comment:
I agree that vector isn't the best choice, but using it here, with first implementation allowed me to ommit multiple definitions of operators for each class. 
We should go for other container when it will be needed to improve our codebase. (refactor)